scripts/methUnite.R

```diff
@@ -0,0 +1,217 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+## Collect arguments
+args <- commandArgs(TRUE)
+
+## Default setting when no arguments passed
+if (length(args) < 1) {
+  args <- c("--help")
+}
+
+## Help section
+if ("--help" %in% args) {
+  cat("
+      Merge methylation samples using methylKit
+
+      Arguments:
+      --inputfiles comma separated input tabix files
+      --sampleids comma separated sample name vector
+      --treatments comma separated treatment vector
+      --assembly genome assembly
+      --context methylation context
+      --destrand wether to merge strands
+      --cores number of processing cores
+      --outdir output directory
+      --suffix suffix for merged file
+      --logFile file to print the logs to
+      --help              - print this text
+
+      Example:
+      ./test.R --arg1=1 --arg2='output.txt' --arg3=TRUE \n\n")
+
+  q(save = "no")
+}
+
+## Parse arguments (we expect the form --arg=value)
+parseArgs <- function(x) strsplit(sub("^--", "", x), "=")
+
+argsDF <- as.data.frame(do.call("rbind", parseArgs(args)))
+argsL <- as.list(as.character(argsDF$V2))
+names(argsL) <- argsDF$V1
+
+# ## catch output and messages into log file
+out <- file(argsL$logFile, open = "wt")
+sink(out, type = "output")
+sink(out, type = "message")
+
+print(argsL)
+
+# Run Functions -----------------------------------------------------------
+
+suppressPackageStartupMessages(library("methylKit"))
+data.table::setDTthreads(8)
+
+## Load variables
+inputs <- strsplit(argsL$inputfiles, ",", fixed = FALSE, perl = FALSE, useBytes = FALSE)[[1]]
+sampleids <- strsplit(argsL$sampleids, ",", fixed = FALSE, perl = FALSE, useBytes = FALSE)[[1]]
+cores <- as.numeric(argsL$cores)
+assembly <- argsL$assembly
+suffix <- argsL$suffix
+outdir <- argsL$outdir
+destrand <- ifelse(tolower(argsL$destrand) %in% c("true", "yes"), TRUE, FALSE)
+
+message("Remapping Treatment Description to number.")
+# split all treatment values, could be numeric or not
+treatmentsStr <- strsplit(argsL$treatments, ",", fixed = FALSE, perl = FALSE, useBytes = FALSE)[[1]]
+# convert into named numeric vector
+treatments <- as.numeric(as.factor(treatmentsStr))
+names(treatments) <- treatmentsStr
+
+message(paste(
+  sort(unique(treatmentsStr)),
+  sort(unique(treatments)),
+  sep = ": ", collapse = "\n"
+))
+
+# convert variables and perform checks
+context <- switch(tolower(argsL$context),
+  cpg = "CpG",
+  chh = "CHH",
+  chg = "CHG",
+  NULL
+)
+
+if (is.null(context)) {
+  stop(
+    "The given context <", argsL$context,
+    "> is not among the supported ones ('CpG','CHG','CHH')"
+  )
+}
+
+if (length(inputs) <= 1) {
+  stop("At least two samples required to perform merging")
+}
+
+
+# ---------------------------------------------------
+st <- system.time({
```

Comment:
I'm not a fan of this wrapping.  You could instead put all the work below in a procedure and then do the system.time thing on that procedure.  I think it's generally better to avoid large blocks of nested code.